src/traits.rs

```diff
@@ -1136,30 +1136,42 @@ macro_rules! impl_to_primitive_float_to_float {
     )
 }
 
+macro_rules! impl_to_primitive_float_to_int_or_uint {
+    ($SrcT:ident, $DstT:ident, $slf:expr) => (
+        {
+            if $slf >= $DstT::MIN as $SrcT && $slf <= $DstT::MAX as $SrcT {
```

Comment:
`as` truncates when casting from `float` to `int`
If you want to include `MAX + 0.1` we could check for `number > (MIN as float) - 1.0 && number < (MAX as float) + 1.0`
I'm not entirely sure of the edge cases there, though. (e.g what happens if the float inaccuracy comes into play)